sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/TextAnalyticsSubscriptionKeyCredential.java

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.textanalytics.models;
+
+/**
+ * Subscription key credential that shared across cognitive services, or restrict to single service.
+ */
+public final class TextAnalyticsSubscriptionKeyCredential {
+    private String subscriptionKey;
+
+    /**
+     * Creates a {@code SharedKeyCredential} model that describes subscription key.
+     *
+     * @param subscriptionKey the subscription key
+     */
+    public TextAnalyticsSubscriptionKeyCredential(String subscriptionKey) {
+        this.subscriptionKey = subscriptionKey;
+    }
+
+    /**
+     * Get the subscription key value.
+     *
+     * @return the subscription key value
+     */
+    public String getSubscriptionKey() {
+        return this.subscriptionKey;
+    }
+
+    /**
+     * Set the subscription key value.
+     *
+     * @param subscriptionKey the subscription key
+     * @return the {@link TextAnalyticsSubscriptionKeyCredential} itself
+     */
+    public TextAnalyticsSubscriptionKeyCredential updateCredential(String subscriptionKey) {
```

Comment:
Also, when is this method called? Is this method expected to be called by the user when they want to update the key? If so, it would be good to include in samples and code snippets.

Reply:
I will add the samples and snippet.